### PR TITLE
[TASK] Disentangle TSconfig conditions from TypoScript conditions

### PR DESCRIPTION
**Pull request by lolli42**
To make more clear which functions and variables are available in TSconfig, the docs now come with an own list.

### CODE REVIEW

File: Documentation/UsingSetting/Conditions.rst [hunk 1]
```diff
@@ -1,85 +1,779 @@
...
 
-For a general discussion about TypoScript conditions, please refer to
-:ref:`TypoScript Syntax Study <t3coreapi:typoscript-syntax-conditions>` of the core API document.
+It is possible to use TypoScript conditions in both User TSconfig and Page TSconfig,
+just as it is done in frontend TypoScript. The list of available variables and
+functions is different, though.
 
-For a list of available conditions, please refer to the
-:ref:`TypoScript Reference <t3tsref:condition-reference>`.
+The symfony expression language tends to throw warnings when sub arrays are
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:23]:
```suggestion
The Symfony expression language tends to throw warnings when sub arrays are
```

File: Documentation/UsingSetting/Conditions.rst [hunk 2]
```diff
@@ -1,85 +1,779 @@
...
 
-It is possible to use TypoScript :ref:`conditions <t3coreapi:typoscript-syntax-conditions>`
-in both User TSconfig and Page TSconfig, just as it is done in TypoScript for templates.
+TSconfig TypoScript conditions offer a way to conditionally change TypoScript based
+on current context. See the :ref:`TypoScript syntax condition chapter <t3coreapi:typoscript-syntax-conditions>`
+for the basic syntax of conditions.
 
-For a general discussion about TypoScript conditions, please refer to
-:ref:`TypoScript Syntax Study <t3coreapi:typoscript-syntax-conditions>` of the core API document.
+It is possible to use TypoScript conditions in both User TSconfig and Page TSconfig,
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:19]:
```suggestion
It is possible to use TypoScript conditions in both user TSconfig and page TSconfig,
```

File: Documentation/UsingSetting/Conditions.rst [hunk 3]
```diff
@@ -1,85 +1,779 @@
...
+
+:aspect:`Variable`
+   tree.pagelayout
+
+:aspect:`Type`
+   Integer / String
+
+:aspect:`Description`
+   Check for the defined backend layout of a page including the inheritance of
+   the field `Backend Layout (subpages of this page)`. Only available in page TSconfig,
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:151]:
```suggestion
   the field :guilabel:`Backend Layout (subpages of this page)`. Only available in page TSconfig,
```

File: Documentation/UsingSetting/Conditions.rst [hunk 4]
```diff
@@ -1,85 +1,779 @@
...
+-------------------------
+
+:aspect:`Variable`
+   backend.user.userGroupList
+
+:aspect:`Type`
+   array
+
+:aspect:`Description`
+   Array of user group ids of the current backend user.
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:357]:
```suggestion
   Array of user group IDs of the current backend user.
```

File: Documentation/UsingSetting/Conditions.rst [hunk 5]
```diff
@@ -1,85 +1,779 @@
...
+   Boolean
+
+:aspect:`Description`
+   True if current user is admin.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # Evaluates to true if current BE-User is administrator
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:294]:
```suggestion
      # Evaluates to true if current backend user is administrator
```

File: Documentation/UsingSetting/Conditions.rst [hunk 6]
```diff
@@ -1,85 +1,779 @@
...
+   Boolean
+
+:aspect:`Description`
+   True if current user is logged in.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # Evaluates to true if an BE-User is logged in
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:317]:
```suggestion
      # Evaluates to true if an backend user is logged in
```

File: Documentation/UsingSetting/Conditions.rst [hunk 7]
```diff
@@ -1,85 +1,779 @@
...
+   Integer
+
+:aspect:`Description`
+   UID of current user.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # Evaluates to true if user uid of current logged in BE-User is equal to 5
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:340]:
```suggestion
      # Evaluates to true if user uid of current logged in backend user is equal to 5
```

File: Documentation/UsingSetting/Conditions.rst [hunk 8]
```diff
@@ -1,85 +1,779 @@
...
+-------------
+
+:aspect:`Variable`
+   typo3.version
+
+:aspect:`Type`
+   String
+
+:aspect:`Description`
+   TYPO3_version (e.g. 9.4.0-dev)
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:502]:
```suggestion
   TYPO3_version (e.g. 12.4.0-dev)
```

File: Documentation/UsingSetting/Conditions.rst [hunk 9]
```diff
@@ -1,85 +1,779 @@
...
+   String
+
+:aspect:`Description`
+   TYPO3_version (e.g. 12.4.0-dev)
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      [typo3.version == "9.5.5"]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:508]:
```suggestion
      [typo3.version == "12.4.0"]
```

File: Documentation/UsingSetting/Conditions.rst [hunk 10]
```diff
@@ -1,85 +1,779 @@
...
+-------------
+
+:aspect:`Variable`
+   typo3.version
+
+:aspect:`Type`
+   String
+
+:aspect:`Description`
+   TYPO3_version (e.g. 12.4.0-dev)
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:502]:
```suggestion
   TYPO3 version (e.g. 12.4.0-dev)
```

File: Documentation/UsingSetting/Conditions.rst [hunk 11]
```diff
@@ -1,85 +1,779 @@
...
+------------
+
+:aspect:`Variable`
+   typo3.branch
+
+:aspect:`Type`
+   String
+
+:aspect:`Description`
+   TYPO3_branch (e.g. 9.4)
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:524]:
```suggestion
   TYPO3 branch (e.g. 12.4)
```

File: Documentation/UsingSetting/Conditions.rst [hunk 12]
```diff
@@ -1,85 +1,779 @@
...
+   String
+
+:aspect:`Description`
+   TYPO3 branch (e.g. 12.4)
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      [typo3.branch == "11.5"]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:530]:
```suggestion
      [typo3.branch == "12.4"]
```

File: Documentation/UsingSetting/Conditions.rst [hunk 13]
```diff
@@ -1,85 +1,779 @@
...
+   String
+
+:aspect:`Description`
+   :php:`$GLOBALS['TYPO3_CONF_VARS']['SYS']['devIPmask']`
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      [typo3.devIpMask == "172.18.0.6"]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:552]:
```suggestion
      [typo3.devIpMask == "203.0.113..6"]
```

Use IP address for documentation: https://datatracker.ietf.org/doc/rfc5737/

File: Documentation/UsingSetting/Conditions.rst [hunk 14]
```diff
@@ -1,85 +1,779 @@
...
+   Boolean
+
+:aspect:`Description`
+   Compares against the current TYPO3 branch.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # True if current version is 11.5.x
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:676]:
```suggestion
      # True if current version is 12.4.x
```

File: Documentation/UsingSetting/Conditions.rst [hunk 15]
```diff
@@ -1,85 +1,779 @@
...
+
+:aspect:`Description`
+   Compares against the current TYPO3 branch.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # True if current version is 12.4.x
+      [compatVersion("11.5")]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:677]:
```suggestion
      [compatVersion("12.4")]
```

File: Documentation/UsingSetting/Conditions.rst [hunk 16]
```diff
@@ -1,85 +1,779 @@
...
+:aspect:`Description`
+   Compares against the current TYPO3 branch.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # True if current version is 12.4.x
+      [compatVersion("12.4")]
+      [compatVersion("11.5.0")]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:678]:
```suggestion
      [compatVersion("12.4.0")]
```

File: Documentation/UsingSetting/Conditions.rst [hunk 17]
```diff
@@ -1,85 +1,779 @@
...
+   Compares against the current TYPO3 branch.
+
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+      # True if current version is 12.4.x
+      [compatVersion("12.4")]
+      [compatVersion("12.4.0")]
+      [compatVersion("11.5.1")]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:679]:
```suggestion
      [compatVersion("12.4.1")]
```

File: Documentation/UsingSetting/Conditions.rst [hunk 18]
```diff
@@ -1,85 +1,779 @@
...
 
-.. code-block:: typoscript
-   :caption: EXT:site_package/Configuration/page.tsconfig
+:aspect:`Example`
+   .. code-block:: typoscript
+      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
 
-   [backend.user.userId == 5]
+      # Site identifier
+      [site("identifier") == "typo395"]
```
Comment (brotkrueml) [inline Documentation/UsingSetting/Conditions.rst:767]:
```suggestion
      [site("identifier") == "my_website"]
```

**Review by brotkrueml — APPROVED** [review]

**Comment by peterkraume** [comment]
@lolli42 are you sure, that TSConfig conditions are `site()` aware?
Today, a colleague stumbled over a problem where `[traverse(site("configuration"), "myCustomProperty") == true]` didn't work but only `[1571 in tree.rootLineParentIds]`.
Funnily, the Page TSConfig backend module showed the correct settings from the condition. But in the modified record, the changes to `TCEFORM` were not applied.

**Pull request merged**